src/ukify/ukify.py

```diff
@@ -824,6 +824,20 @@ def make_uki(opts):
     if pcrpkey is None:
         if opts.pcr_public_keys and len(opts.pcr_public_keys) == 1:
             pcrpkey = opts.pcr_public_keys[0]
+            # If we are getting a certificate when using an engine, we need to convert it to public key format
+            if opts.signing_engine is not None:
+                from cryptography.hazmat.primitives import serialization
+                from cryptography.x509 import load_pem_x509_certificate
+
+                try:
+                    cert = load_pem_x509_certificate(pathlib.Path(pcrpkey).read_bytes())
+                    pcrpkey = cert.public_key().public_bytes(
+                        encoding=serialization.Encoding.PEM,
+                        format=serialization.PublicFormat.SubjectPublicKeyInfo,
+                    )
+                except ValueError:
+                    # Not an X.509 certificate, nothing to do
+                    pass
```

Comment:
We should probably fail here. We explicitly document that we should receive an x509 cert when a signing engine is passed.

Reply:
It's not that specific and left quite open:

```
  --pcr-public-key PATH
                        public part of the keypair or engine-specific designation for signing PCR signatures
```

Who knows what kind of engines are out there doing weird things?

Reply:
@bluca Written by yourself:

>  If <varname>SigningEngine=</varname>/<option>--signing-engine=</option> is specified, then the private keys arguments will be passed verbatim to OpenSSL as URIs, and the public key arguments will be loaded as X.509 certificates, so that signing can be performed with an OpenSSL engine.

So either we change the documentation or we should fail if we're not passed a X509 certificate.

Reply:
Ah that's in sd-measure, forgot about that. How dare you holding the nonsense I write against me!